client/core/core.go

```diff
@@ -1936,24 +1998,50 @@ func (c *Core) ChangeAppPass(appPW, newAppPW []byte) error {
 // ReconfigureWallet updates the wallet configuration settings, it also updates
 // the password if newWalletPW is non-nil. Do not make concurrent calls to
 // ReconfigureWallet for the same asset.
-func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg map[string]string) error {
+func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, form *WalletForm) error {
 	crypter, err := c.encryptionKey(appPW)
 	if err != nil {
 		return newError(authErr, "ReconfigureWallet password error: %v", err)
 	}
+	assetID := form.AssetID
+	walletDef, err := walletDefinition(assetID, form.Type)
+	if err != nil {
+		return err
+	}
 	oldWallet, found := c.wallet(assetID)
 	if !found {
 		return newError(missingWalletErr, "%d -> %s wallet not found",
 			assetID, unbip(assetID))
 	}
 	oldDepositAddr := oldWallet.currentDepositAddress()
 	dbWallet := &db.Wallet{
+		Type:        form.Type,
 		AssetID:     oldWallet.AssetID,
-		Settings:    cfg,
+		Settings:    form.Config,
 		Balance:     &db.Balance{}, // in case retrieving new balance after connect fails
 		EncryptedPW: oldWallet.encPW(),
 		Address:     oldDepositAddr,
 	}
+
+	if walletDef.Seeded {
+		if len(newWalletPW) > 0 {
+			return newError(passwordErr, "cannot set a password on a seeded wallet")
+		}
+
+		exists, err := asset.WalletExists(assetID, form.Type, c.assetDataDirectory(assetID), form.Config, c.net)
```

Comment:
The btc driver implementation of this method (haven't checked others) expects that this method may be called for RPC wallets as well. Is that the case or is that planned?